core/services/gateway/handlers/functions/allowlist/allowlist.go

```diff
@@ -219,6 +219,17 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 		if err != nil {
 			return errors.Wrap(err, "error calling GetAllAllowedSenders")
 		}
+
+		err = a.orm.PurgeAllowedSenders()
```

Comment:
This step to purge the DB will happen on every sync of the stored allow list right?

What is the performance impact of this to delete all and re-add? Because on the new version of the ToS contract this is not needed. Can we put purging behind a feature flag?

Reply:
this will only happen for the old ToS contracts that will `GetAllAllowedSenders`. 

I like your idea of checking the contract typeAndVersion instead of a job spec change. Also the naming of the feature flag does not make sense now because its storing anyways. thanks will change that